### PR TITLE
zh: update architecture; add new features in DM

### PR DESCRIPTION
**Pull request by csuzhangxc**
### What is changed, added or deleted? (Required)

- update architecture
- add new features in DM

### Which DM version(s) do your changes apply to? (Required)

- [x] master (the latest development version, including v2.0 changes for now)
- [ ] v1.0 (TiDB DM 1.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [x] Have version specific changes 
- [ ] Might cause conflicts

### CODE REVIEW

**Comment by csuzhangxc** [comment]
@GMHDBJD @TomShawn PTAL

File: zh/_index.md [hunk 1]
```diff
@@ -8,6 +8,14 @@ aliases: ['/docs-cn/tidb-data-migration/dev/']
 
 [TiDB Data Migration](https://github.com/pingcap/dm) (DM) 是一体化的数据同步任务管理平台，支持从 MySQL 或 MariaDB 到 TiDB 的全量数据迁移和增量数据同步。使用 DM 工具有利于简化错误处理流程，降低运维成本。
 
+DM 2.0 相比于 1.0，主要有以下改进：
+
+- 数据迁移任务的高可用，部分 DM-master、DM-worker 节点异常后仍能保证数据迁移任务的正常运行。
+- 新的 [乐观协调模式的 shard DDL](feature-shard-merge-optimistic.md) 可以在部分场景下减少 shard DDL 同步过程中的延迟、支持上游数据库灰度变更等场景。
```
Comment (TomShawn) [inline zh/_index.md:14]:
```suggestion
- 新的[乐观协调模式的 shard DDL](feature-shard-merge-optimistic.md) 可以在部分场景下减少 shard DDL 同步过程中的延迟、支持上游数据库灰度变更等场景。
```

File: zh/_index.md [hunk 2]
```diff
@@ -8,6 +8,14 @@ aliases: ['/docs-cn/tidb-data-migration/dev/']
...
+- 更好的易用性，包括新的[错误处理机制](handle-failed-sql-statements.md) 及更清晰易读的错误信息与错误处理建议。
```
Comment (TomShawn) [inline zh/_index.md:15]:
```suggestion
- 更好的易用性，包括新的[错误处理机制](handle-failed-sql-statements.md)及更清晰易读的错误信息与错误处理建议。
```

File: zh/overview.md [hunk 3]
```diff
@@ -7,6 +7,14 @@ aliases: ['/docs-cn/tidb-data-migration/dev/overview/','/docs-cn/tools/dm/overvi
 
 [TiDB Data Migration](https://github.com/pingcap/dm) (DM) 是一体化的数据同步任务管理平台，支持从 MySQL 或 MariaDB 到 TiDB 的全量数据迁移和增量数据同步。使用 DM 工具有利于简化错误处理流程，降低运维成本。
 
+DM 2.0 相比于 1.0，主要有以下改进：
+
+- 数据迁移任务的高可用，部分 DM-master、DM-worker 节点异常后仍能保证数据迁移任务的正常运行。
+- 新的 [乐观协调模式的 shard DDL](feature-shard-merge-optimistic.md) 可以在部分场景下减少 shard DDL 同步过程中的延迟、支持上游数据库灰度变更等场景。
```
Comment (TomShawn) [inline zh/overview.md:13]:
```suggestion
- 新的[乐观协调模式的 shard DDL](feature-shard-merge-optimistic.md) 可以在部分场景下减少 shard DDL 同步过程中的延迟、支持上游数据库灰度变更等场景。
```

File: zh/overview.md [hunk 4]
```diff
@@ -7,6 +7,14 @@ aliases: ['/docs-cn/tidb-data-migration/dev/overview/','/docs-cn/tools/dm/overvi
...
+- 更好的易用性，包括新的[错误处理机制](handle-failed-sql-statements.md) 及更清晰易读的错误信息与错误处理建议。
```
Comment (TomShawn) [inline zh/overview.md:14]:
```suggestion
- 更好的易用性，包括新的[错误处理机制](handle-failed-sql-statements.md)及更清晰易读的错误信息与错误处理建议。
```

**Review by GMHDBJD — COMMENTED** [review]
LGTM

File: zh/_index.md [hunk 5]
```diff
@@ -8,6 +8,14 @@ aliases: ['/docs-cn/tidb-data-migration/dev/']
 
 [TiDB Data Migration](https://github.com/pingcap/dm) (DM) 是一体化的数据同步任务管理平台，支持从 MySQL 或 MariaDB 到 TiDB 的全量数据迁移和增量数据同步。使用 DM 工具有利于简化错误处理流程，降低运维成本。
 
+DM 2.0 相比于 1.0，主要有以下改进：
+
+- 数据迁移任务的高可用，部分 DM-master、DM-worker 节点异常后仍能保证数据迁移任务的正常运行。
+- 新的[乐观协调模式的 shard DDL](feature-shard-merge-optimistic.md) 可以在部分场景下减少 shard DDL 同步过程中的延迟、支持上游数据库灰度变更等场景。
```
Comment (TomShawn) [inline zh/_index.md:14]:
```suggestion
- [乐观协调模式下的 sharding DDL](feature-shard-merge-optimistic.md) 可以在部分场景下减少 sharding DDL 同步过程中的延迟、支持上游数据库灰度变更等场景。
```

File: zh/_index.md [hunk 6]
```diff
@@ -8,6 +8,14 @@ aliases: ['/docs-cn/tidb-data-migration/dev/']
...
+- 更好的易用性，包括新的[错误处理机制](handle-failed-sql-statements.md)及更清晰易读的错误信息与错误处理建议。
+- 与上下游数据库及 DM 各组件间连接的 [TLS 支持](enable-tls.md)。
+- 实验性支持从 MySQL 8.0 迁移数据。
```
Comment (TomShawn) [inline zh/_index.md:17]:
```suggestion
- 实验性地支持从 MySQL 8.0 迁移数据。
```

File: zh/overview.md [hunk 7]
```diff
@@ -7,6 +7,14 @@ aliases: ['/docs-cn/tidb-data-migration/dev/overview/','/docs-cn/tools/dm/overvi
 
 [TiDB Data Migration](https://github.com/pingcap/dm) (DM) 是一体化的数据同步任务管理平台，支持从 MySQL 或 MariaDB 到 TiDB 的全量数据迁移和增量数据同步。使用 DM 工具有利于简化错误处理流程，降低运维成本。
 
+DM 2.0 相比于 1.0，主要有以下改进：
+
+- 数据迁移任务的高可用，部分 DM-master、DM-worker 节点异常后仍能保证数据迁移任务的正常运行。
+- 新的[乐观协调模式的 shard DDL](feature-shard-merge-optimistic.md) 可以在部分场景下减少 shard DDL 同步过程中的延迟、支持上游数据库灰度变更等场景。
```
Comment (TomShawn) [inline zh/overview.md:13]:
```suggestion
- [乐观协调模式下的 sharding DDL](feature-shard-merge-optimistic.md) 可以在部分场景下减少 sharding DDL 同步过程中的延迟、支持上游数据库灰度变更等场景。
```

File: zh/overview.md [hunk 8]
```diff
@@ -7,6 +7,14 @@ aliases: ['/docs-cn/tidb-data-migration/dev/overview/','/docs-cn/tools/dm/overvi
...
+- 更好的易用性，包括新的[错误处理机制](handle-failed-sql-statements.md)及更清晰易读的错误信息与错误处理建议。
+- 与上下游数据库及 DM 各组件间连接的 [TLS 支持](enable-tls.md)。
+- 实验性支持从 MySQL 8.0 迁移数据。
```
Comment (TomShawn) [inline zh/overview.md:16]:
```suggestion
- 实验性地支持从 MySQL 8.0 迁移数据。
```

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**